src/index.ts

```diff
@@ -0,0 +1,58 @@
+import { PrismaClient } from '@prisma/client';
+import express, { type Request } from 'express';
+import { ZenStackMiddleware } from '@zenstackhq/server/express';
+import { withPresets } from '@zenstackhq/runtime';
+
+// Let's use this project to collaborate
+
+const prisma = new PrismaClient();
+const app = express();
+
+app.use(express.json());
+
+function getSessionUser(request: Request) {
+    // TODO: Get user id from request
+    return 'user1';
+}
+
+app.use(
+    '/api',
+    ZenStackMiddleware({
+        getPrisma: async (request) => {
+            const uid = getSessionUser(request);
+
+            // fetch full user from db
+            const dbUser = await prisma.user.findUnique({ where: { id: uid } });
+
+            // find the group with global post permission for the current user
+            const memberOfPostAdminGroup = await prisma.groupUser.findFirst({
+                include: { group: true },
+                where: {
+                    userId: uid,
+                    group: {
+                        postPermission: {
+                            not: null,
+                        },
+                    },
+                },
+            });
+
+            const user = {
+                id: uid,
+                postPermission: dbUser?.postPermission,
+                groupPostPermission: memberOfPostAdminGroup?.group.postPermission,
+            };
+            withPresets(prisma, { user });
```

Comment:
The logic here precomputed user's global permission to Post and also if the user is a member of a group with global Post permission. The result is added to the context and is available from `auth()`.

Reply:
Yeah, this is a nice way to make custom permissions